File: drools-spring-app/src/main/java/travel_recommendation/service/interfaces/DestinationService.java
package travel_recommendation.service.interfaces;

import travel_recommendation.dto.LikeDto;
import travel_recommendation.dto.TravelDto;
import travel_recommendation.model.*;
import travel_recommendation.model.enums.DestinationType;
import travel_recommendation.model.enums.TransportationType;
import travel_recommendation.model.enums.Weather;

import java.util.List;

public interface DestinationService {
    List<Destination> getDestinationList(String username, TransportationType transportationType, double budget,
                                         DestinationType destinationType, Weather weather, String continent);

    String like(LikeDto like);

    void reserve(TravelDto travelDto);

}

File: client-app/src/app/services/destination-service/destination.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Like } from '../../model/like-model';
import { Travel } from '../../model/travel-model';
import {User} from '../../model/user-model';
import { TravelDto } from 'src/app/dto/travel-dto';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class DestinationService {
  baseUrl: String = environment.appUrl + 'destination';
  headers = new HttpHeaders({'Content-Type' : 'application/json',
                             'Authorization' : `Bearer ${localStorage['jwt']}`});

  constructor(private http: HttpClient) { }

  findDestinations(transportationType: String, budget: Number, destinationType: String,
    weather: String, continent: String) {
    return this.http.get(this.baseUrl + '?transportationType='+transportationType+
    '&budget='+budget+'&destinationType='+destinationType+'&weather='+weather+'&continent='+continent, { headers: this.headers });
  }

  like(like: Like) {
    return this.http.post(this.baseUrl + '/like', like, { headers: this.headers });
  }
  reserve(travel: TravelDto) {
    return this.http.post(this.baseUrl + '/reservation', travel, { headers: this.headers });
  }
}
